### PR TITLE
Remove Gateway in Major Code Paths

### PR DESCRIPTION
**Pull request by longbowlu**
This PR removes majority dependencies of gateway from the rest of the system. In particular, this PR has the following changes: 
1. Embedded gateway is the only presence of gateway in non-gateway specific paths (the integrations tests for gateway server are gateway specific paths). The only purpose is to keep gateway integration tests happy. Once we roll out this change and see it stabilizes, we will remove both. 
2. introduce `TestClusterBuilder` to build local test cluster, consolidate existing test network creation methods 
3. introduce `TransactionExecutionResult` to consolidate `SuiTransactionResponse` (embedded gateway api) and `SuiExecuteTransactionResponse` (from fullnode execution api)
4. in sui-sdk, add basic polling when false `confirmed_local_execution` is returned from `WaitForLocalExecution` results
5. in api docs, use `FullNodeTransactionExecutionApi` and remove `RpcGatewayImpl`. This updates docs for api `sui_executeTransaction` 
6. replace `http://gateway.devnet.sui.io/` with `http://fullnode.devnet.sui.io/`
7. remove `gateway_address` in sui-test-validator
8. only allow state sync in embedded gateway client. 

what else needs to be done before we shut down the gateway server?
- [x] sui client (cli) is good to go. Now it connects to devnet fullnode by default.
- [x] rust sdk is good to go
- [ ] ts sdk needs updates
- [ ]  wallet needs updates
- [x] explorer is probably fine, needs confirmation
- [ ] community sdk (need to update community about this potential breaking change)
- [ ] add transaction builder apis in fullnode so this transition is seamless (mostly for non-rust sdks/dapps)

What needs to be done after shutting down gateway server:
1. a few things still rely on gatway.yaml e.g. stress. Replace with network.yaml 
2. decouple some dependencies in GatewayState, except the embedded gateway - the remaining parts are small
3. which parts in GatewayState to move into SDK?
4. whether/when to fully remove the embedded gateway?

### CODE REVIEW

**Comment by patrickkuo** [comment]
> 4. whether/when to fully remove the embedded gateway?
I think we should remove embedded gateway, embedded gateway was created at the beginning (before gateway and fullnode rpc server) to make it easier to connect to the local network. Now we have fullnode we can default the CLI to connect to the fullnode RPC by default, we can also consider starting the fullnode in `sui start` to make local testing easier.

**Comment by patrickkuo** [comment]
We can also merge `FullnodeApi` and `ReadApi` after this PR

**Comment by longbowlu** [comment]
> > 4. whether/when to fully remove the embedded gateway?
> 
> I think we should remove embedded gateway, embedded gateway was created at the beginning (before gateway and fullnode rpc server) to make it easier to connect to the local network. Now we have fullnode we can default the CLI to connect to the fullnode RPC, we can also consider starting the fullnode in `sui start` to make local testing easier.

are there any good stuff we want to move from embedded gateway to sui-sdk?

**Review by patrickkuo — APPROVED** [review]
Great work! LGTM.

I think we should remove embedded gateway and start a fullnode when running `sui start`, I can help pick that up!

File: crates/sui-open-rpc/spec/openrpc.json [hunk 1]
```diff
@@ -227,13 +227,21 @@
...
             "$ref": "#/components/schemas/Base64"
           }
+        },
+        {
+          "name": "request_type",
+          "description": "The request type",
+          "required": true,
+          "schema": {
+            "$ref": "#/components/schemas/ExecuteTransactionRequestType"
+          }
```
[the remark this answers is not in the document]
Comment (longbowlu) [reply crates/sui-open-rpc/spec/openrpc.json:237]: Is it safe to land this PR and later update TS SDK? I think this file is only for documentation

[the remark this answers is not in the document]
Comment (patrickkuo) [reply crates/sui-open-rpc/spec/openrpc.json:237]: Ya this is just a documentation, but changes to this file indicates there is a schema changes in json rpc which might cause error in TS SDK.

[the remark this answers is not in the document]
Comment (666lcz) [reply crates/sui-open-rpc/spec/openrpc.json:237]: Yes this is safe to land - TS SDK does not check for the schema directly. This TS change for this endpoint has been implemented: https://github.com/MystenLabs/sui/blob/93f796a5cce03ea1abb62904b28263b297c856c7/sdk/typescript/src/providers/json-rpc-provider.ts#L385

**Pull request merged**